packages/jest-core/src/collectHandles.ts

```diff
@@ -141,14 +132,18 @@ export default function collectHandles(): HandleCollectionResult {
     // For example, Node.js TCP Servers are not destroyed until *after* their
     // `close` callback runs. If someone finishes a test from the `close`
     // callback, we will not yet have seen the resource be destroyed here.
-    await asyncSleep(100);
+    await asyncSleep(0);
 
     if (activeHandles.size > 0) {
-      // For some special objects such as `TLSWRAP`.
-      // Ref: https://github.com/facebook/jest/issues/11665
-      runGC();
+      await asyncSleep(30);
```

Comment:
Checking length is free, let's check it first. Another 100ms might be a bit long? I modified it to 100ms, you can also modify it to 0-10ms if you want, lol.

Reply:
as low as possible while still passing all tests (without introducing flake) seems ideal 😀